File: pa3/DictionaryTest.java
// --------------------------------------------------------------------------
// DictionaryTest.java
// Matthew Tan
// mxtan
// pa3
// CMPS 12B/M
// --------------------------------------------------------------------------

public class DictionaryTest
{
    public static void main(String[] args)
    {
        String test;
        Dictionary A = new Dictionary();

        // tests inserting items into Dictionary ADT
        A.insert("1", "a");
        A.insert("2", "b");
        A.insert("3", "c");
        A.insert("4", "d");
        A.insert("5", "e");
        A.insert("6", "f");
        A.insert("7", "g");
        System.out.println(A);
        
        // tests looking up items in Dictionary ADT
        test = A.lookup("1");
        System.out.println("key = 1 " + (test == null ? "not found" 
             : ("value = " + test)));

        test = A.lookup("3");
        System.out.println("key = 3 " + (test == null ? "not found" 
             : ("value = " + test)));

        test = A.lookup("5");
        System.out.println("key = 5 " + (test == null ? "not found" 
             : ("value = " + test)));

        test = A.lookup("7");
        System.out.println("key = 7 " + (test == null ? "not found" 
             : ("value = " + test)));

        test = A.lookup("8");
        System.out.println("key = 8 " + (test == null ? "not found" 
             : ("value = " + test)));
        System.out.println();

        // Following causes DuplicateKeyException
        // A.insert("2", "f");

        // Following causes KeyNotFoundException
        // A.delete("8");

        // You can do the following:
        // A.insert("9", "i");
        // insert works because you
        // are inserting a new key that is not duplicate
        // System.out.println(A);
       
        // tests deleting items in DictionaryADT
        A.delete("1");
        A.delete("3");
        A.delete("7");
        System.out.println(A);
                
        System.out.println(A.isEmpty());
        System.out.println(A.size());
        A.makeEmpty();

        // prints out an empty list:
        // System.out.println(A);
        System.out.println(A.isEmpty());

        // Causes KeyNotFoundException since 
        // there are no keys in the list
        // A.delete("4");

        System.out.println(A);
    }
}

File: lab5/DictionaryTest.c
//---------------------------------------------------------------------------
// DictionaryTest.c
// Matthew Tan
// mxtan
// lab5
// CMPS 12B/M
//---------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

int main(int argc, char* argv[])
{
    Dictionary A = newDictionary();
 
    insert(A, "1", "a");
    insert(A, "2", "b");
    insert(A, "3", "c");
    insert(A, "4", "d");
    insert(A, "5", "e");
    insert(A, "6", "f");
    insert(A, "7", "g");

    // insert(A, "2", "c"); // causes a Duplicae Key Error
    printf("Before delete: start printDictionary--------------\n");
    printDictionary(stdout, A);
    printf("Is the list empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Current size is: %d\n", size(A));
    char* test;
    test = lookup(A, "1");
    printf("value of lookup: %s\n", test);
    printf("Before delete: end printDictionary--------------\n");
    delete(A, "1");
    delete(A, "3");
    delete(A, "5");
    delete(A, "7");
    printDictionary(stdout, A);
    printf("Is the current list empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Current size is: %d\n", size(A));
    makeEmpty(A);
    printf("After makeEmpty: start printDictionary-------------\n");
    printDictionary(stdout, A);
    // delete(A, "1"); // causes an Empty Dictionary Error
    // test = lookup(A, "1"); // causes a Empty Dictionary Error
    printf("After makeEmpty: end printDictionary-------------\n");
    printf("Is the current list empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Current size: %d\n", size(A));
}
